interfaces/builtin/upower_observe.go

```diff
@@ -20,7 +20,12 @@
 package builtin
 
 import (
+	"bytes"
+	"fmt"
+
 	"github.com/snapcore/snapd/interfaces"
+	"github.com/snapcore/snapd/release"
+	"github.com/snapcore/snapd/snap"
 )
 
 const upowerObserveConnectedPlugAppArmor = `
```

Comment:
I previously said 'LGTM' but that was premature. upower-observe allows a for the plug to connect to the slot, but there is no PermanentSlot policy or ConnectedSlot policy for this interface. Based on this interface and the upower-control PR (https://github.com/snapcore/snapd/pull/2498), it appears you are trying to do something like with location-observe and location-control where the 'observe' interface allows GPS tracking (eg, reading) but the 'control' interface allows configuring the location service. Is this what you are planning for upower-observe and upower-control (where 'observe' has some safe subset of APIs and 'control' allows something else)? If so, you need to implement both interfaces with PermanentSlot and ConnectedSlot. If not, this should all be in upower-observe I think.